configs/William/Msg_Map_process_test.py

```diff
@@ -0,0 +1,212 @@
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+""" This file creates a barebones system and executes 'hello', a simple Hello
+World application. Adds a simple memobj between the CPU and the membus.
+
+This config file assumes that the x86 ISA was built.
+"""
+
+# import the m5 (gem5) library created when gem5 is built
+import m5
+# import all of the SimObjects
+from m5.objects import *
+from m5.util.convert import *
+from caches import *
+
+# import the SimpleOpts module
+from common import SimpleOpts
+import math
+
+import argparse
+
+
+parser = argparse.ArgumentParser()
+
+
+# only used with traffic generator
+options = parser.parse_args()
+
+options.rd_perc = 35
+options.block_size = 64
+options.num_chnls = 1
+options.duration = int(toLatency("100us") * 1e12)
+options.min_addr = 0
+#options.min_addr = toMemorySize(str(1 * options.num_chnls) + 'MB')
+options.max_addr = toMemorySize(str(4 * options.num_chnls) + 'KiB')
+options.max_addr = toMemorySize(str(1024 * options.num_chnls) + 'MB')
+options.injection_rate = 1
+injection_period = int((1e12 * options.block_size) /
+                    (options.injection_rate * 1073741824))
+options.min_period = injection_period
+options.max_period = injection_period
+
+
+
+# create the system we are going to simulate
+system = System()
+
+# Set the clock fequency of the system (and all of its children)
+system.clk_domain = SrcClockDomain()
+system.clk_domain.clock = '1GHz'
+system.clk_domain.voltage_domain = VoltageDomain()
+
+# Set up the system
+system.mem_mode = 'timing'               # Use timing accesses   AddrRange('1023MB', '1047556KiB')]
+system.mem_ranges = [AddrRange(0, '255MB'), AddrRange('255MB', '256MB')] # Create an address range
+# system.mem_ranges = [AddrRange(0, '1023MB'), AddrRange('1023MB', '1024MB')] # Create an address range
+
+# system.mem_ranges = [AddrRange(0, '1MB'), AddrRange('1MB', '1024MB')] # Create an address range
+
+addr_range = system.mem_ranges[0]
+queue_ranges = [system.mem_ranges[1]]
+
+# system.membus = SystemXBar(width = 64, max_routing_table_size = 16777216)
+system.membus = IOXBar(width = 64)
```

Comment:
The membus (i.e., the point of coherence) can't be an `IOXBar`, it must be a `SystemXBar`